spec/Request/user_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+# test controlers
+
+RSpec.describe UsersController, type: :request do
+  describe 'GET #index' do
+    before(:example) do
+      get '/users'
+    end
+
+    context 'valid Authorization header' do
+      it 'it renders the index template' do
+        expect(response).to render_template('index')
+      end
+    end
+
+    context 'expect response to be succesful' do
+      it 'should return 200' do
+        expect(response).to have_http_status(:ok)
+      end
+    end
+
+    context 'expect to return placeholder' do
+      it ' should return place holder' do
+        # get :index
+        expect(response.body).to include('Users List')
+      end
+    end
+  end
+end
+
+# Path: spec/Request/user_spec.rb
+
+RSpec.describe UsersController, type: :request do
+  describe 'GET show' do
+    before(:example) do
+      get '/users/7'
```

Comment:
- [ ] 9 of your tests are failing, kindly update your tests. Hint: you shouldn't hardcore your users or posts on your tests.
![image](https://user-images.githubusercontent.com/62705947/197796041-beda5969-5d2b-4289-b11b-5add871b1c16.png)

Reply:
@GzimAsani  Thank you for reviewing my project, I will make the necessary corrections as suggested